src/server/game/Entities/GameObject/GameObject.cpp

```diff
@@ -2490,8 +2490,16 @@ void GameObject::SetLootState(LootState state, Unit* unit)
     AI()->OnLootStateChanged(state, unit);
 
     // Start restock timer if the chest is partially looted or not looted at all
-    if (GetGoType() == GAMEOBJECT_TYPE_CHEST && state == GO_ACTIVATED && GetGOInfo()->chest.chestRestockTime > 0 && m_restockTime == 0)
-        m_restockTime = GameTime::GetGameTime() + GetGOInfo()->chest.chestRestockTime;
+    if (GetGoType() == GAMEOBJECT_TYPE_CHEST && state == GO_ACTIVATED)
+    {
+        GameObjectTemplate const* goInfo = GetGOInfo();
+        if (goInfo->chest.chestRestockTime > 0 && m_restockTime == 0)
+            m_restockTime = GameTime::GetGameTime() + goInfo->chest.chestRestockTime;
+
+        // If world chests were opened, despawn them after 5 minutes
+        if (goInfo->chest.chestRestockTime == 0 && m_restockTime == 0 && GetMap())
```

Comment:
whats the reason for the GetMap() check here?

Reply:
Should be removed, in our fork we only let this run on world maps (continents) and looks like this was an oversight in porting.

Reply:
yeah well while that check is still needed I remove it because it won't build otherwise, that's why I requested for some help with it.

Reply:
I see. Can either port this: https://github.com/Project-Epoch/TrinityCore/blob/7f8c54e0a10f39b0164ef3914f4697e017988482/src/server/game/Maps/Map.cpp#L4599

Or maybe `! GetMap()->Instanceable()` will suffice here.

Reply:
the negate GetMap()->Instanceable() didn't worked there so I falled to other option, ty for your input.